### ISSUE TITLE
Page is crawled multiple times if link casing varies

### ISSUE THREAD

**Issue by kurthamilton** [sjdirect/abot#111]
If the same page is linked to twice with different casing, that page is crawled (or at least logged) twice.

E.g: These pages are considered unique, whereas I think they should only be crawled once.

/Page.aspx?Param=Value
/page.aspx?param=value

**Comment by LeMoussel** [comment]
These pages are considered unique, I think that it is normal that they are twice crawled.

**Comment by kurthamilton** [comment]
OK, thanks for the confirmation.

**Comment by Ruffio** [comment]
But should they/it be crawled twice? Does a webserver distinguish between casing? If not, then I don't think that it should be crawled twice as the second time will be unnecessary, as then content will be the same as first time. IMHO.

**Comment by LeMoussel** [comment]
But your are not sure that content will be the same as first time, it may be different. Therefore, I think it is better to do twice crawled. (Rem: Google bot crawler, crawl this 2 Urls => Duplicate content for Google)

**Comment by kurthamilton** [comment]
I only have experience of IIS, which is case insensitive (hence my initial comment). However, this Stack Exchange answer explains that different web servers behave differently, and that the application may also choose to interpret urls in their own way: 
[http://webmasters.stackexchange.com/questions/69705/should-url-paths-be-case-sensitive](http://webmasters.stackexchange.com/questions/69705/should-url-paths-be-case-sensitive).

I think that configurable case sensitivity would be a nice option, if I as a user only cares about unique content rather than unique links.

But having said that, good SEO dictates that links should match case, so it is useful to see where casing differs.

**TL;DR;** I agree that links should be case sensitive. Links of different case in my [inherited] application are my problem as a developer.

**Comment by Ruffio** [comment]
Ahh, nice to know. I also only have experience with IIS. I agree in that an option to ignore casing would be nice to have.

**Comment by sjdirect** [comment]
Like mentioned above, the two links could be different and treating them as the same (even if configurable) is technically incorrect. You can however make that determination on your own with a custom implementation of ICrawledPageRepository.
